src/components/edit_account/EditAccountSettings.js

```diff
@@ -0,0 +1,432 @@
+                this.setState({
+                    'newEmailReentryIsValid': undefined
+                });
+            } else if (this.validateEmailReentry(value)){
+                this.setState({
+                    'newEmailReentryIsValid': true
+                });
+            } else {
+                this.setState({
+                    'newEmailReentryIsValid': undefined
+                });
+            }
+            break;
+        default:
+            break;
+        }
+        this.setState({
+            [event.target.name]: value,
+        });
+    }
+
+    /**
+     * Update component state when user unfocus from the input field
+     * @param {Event} event 
+     */
+    handleOnBlur(event) {
+        switch (event.target.name) {
+        case 'newPassword':
+            if (this.state.newPasswordIsValid !== undefined 
+                    || this.state.newPassword.length > 0){
+                this.setState({
+                    'newPasswordIsValid': this.validatePassword(this.state.newPassword)
+                });
+            } else {
+                this.setState({
+                    'newPasswordIsValid': undefined
+                });
+            }
+            break;
+        case 'newPasswordReentry':
+            if (this.state.newPasswordReentry.length === 0){
+                this.setState({
+                    'newPasswordReentryIsValid': undefined
+                });
+            } else {
+                this.setState({
+                    'newPasswordReentryIsValid': this.validatePasswordReentry()
+                });
+            }
+            break;
+        case 'newEmail':
+            if (this.state.newEmail.length === 0){
+                this.setState({
+                    'newEmailIsValid': undefined
+                });
+            } else{
+                this.setState({
+                    'newEmailIsValid': this.validateEmail(this.state.newEmail)
+                });
+            }
+            break;
+        case 'newEmailReentry':
+            if (this.state.newEmailReentry.length === 0){
+                this.setState({
+                    'newEmailReentryIsValid': undefined
+                });
+            } else{
+                this.setState({
+                    'newEmailReentryIsValid': this.validateEmailReentry(this.state.newEmailReentry)
+                });
+            }
+            break;
+        default:
+            break;
+        }
+    }
+
+    /**
+     * Set value to element state
+     * @param {String} element 
+     * @param {*} value 
+     */
+    handleSetState(element, value) {
+        this.setState({[element] : value});
+    }
+
+    /**
+     * Return true if current password is valid
+     */
+    validateCurrentPassword() {
+        return true;
+    }
+
+    /**
+     * Returns true if the password is valid and false otherwise
+     * @param {String} password
+     */
+    validatePassword(password) {
+        if (password.length < 8) {
```

Comment:
Please don't duplicate code like this. Perform the necessary refactoring so you don't have to do this.